packages/compass-crud/src/components/document-list.tsx

```diff
@@ -263,6 +263,12 @@ class DocumentList extends React.Component<DocumentListProps> {
     void this.props.store.runBulkDelete();
   }
 
+  onExportToLanguageDeleteQuery() {
+    void this.props.store.openDeleteQueryExportToLanguageDialog(
+      toJSString(this.props.store.state.query.filter) || '{}'
+    );
```

Comment:
If this only depends on store values, why is this logic implemented in the component?

Reply:
You are right, I'm still refactoring a bit. My idea was to remove the parameter from the action and get directly the data from the store.